contrib/registry/consul/client.go

```diff
@@ -205,31 +244,23 @@ func (c *Client) Register(_ context.Context, svc *registry.ServiceInstance, enab
 			defer ticker.Stop()
 			for {
 				select {
-				case <-c.ctx.Done():
-					_ = c.cli.Agent().ServiceDeregister(svc.ID)
-					return
-				default:
-				}
-				select {
-				case <-c.ctx.Done():
+				case <-cc.ctx.Done():
```

Comment:
This part of the code will cause a secondary logout, but it seems to have no actual impact